### PR TITLE
DIS-43 SortKey and SortDirection for Thread and Post lists

### PR DESCRIPTION
**Pull request by jogura**
Use two separate parameters to define the sort key and the direction (ascending, descending).

* Modify the resources, services and DAOs
* Update the tests 

@jimmordino @armonr

### CODE REVIEW

File: service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java [hunk 1]
```diff
@@ -138,12 +152,24 @@ public Response getForum(
...
+    try {
+      validateSortParams(sortKey, sortDirection);
+    } catch (Exception ex) {
+      DiscussionServiceApplication.LOGGER.info("Invalid sort params - " + ex.getMessage());
+      return ErrorResponseBuilder.buildErrorResponse(
+          ex.getMessage(),
+          Response.Status.BAD_REQUEST,
+          uriInfo.getRequestUri());
+    }
+
```
Comment (jogura) [inline service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java]: I will refactor this and move this functionality into the the `SortKey` in another branch

File: service/discussion/src/test/java/com/wikia/discussionservice/services/ForumServiceTest.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public void getForumRootWithNestedChildren() {
     Forum childForum5  = createForum(5L, 2L);
     forums.add(childForum5);
 
-    Forum childForum6  = createForum(6L, 4L);
+    Forum childForum6  = createForum  (6L, 4L);
```
Comment (jogura) [inline service/discussion/src/test/java/com/wikia/discussionservice/services/ForumServiceTest.java]: minor: erroneous whitespace will remove in another branch

File: service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java [hunk 3]
```diff
@@ -255,4 +281,15 @@ private Response notFound(String errorMessage) {
             .title(errorMessage)
             .build());
   }
+
+  private void validateSortParams(SortKey sortKey, SortDirection sortDirection) {
+    Preconditions.checkNotNull(sortKey, "Invalid sort key");
+
+    Preconditions.checkNotNull(sortDirection, "Invalid sort direction");
```
Comment (jogura) [inline service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java]: These preconditions will never be null at this point. If an invalid sort key or direction is passed in a `NoSuchElementException` will be thrown when we try to lookup the enum.

Comment (armonr) [inline service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java]: Good catch. We also need to fix the validation of responseGroups as well by the same token.

File: service/discussion/src/main/java/com/wikia/discussionservice/enums/SortKey.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+/**
+ * SortKey represents the sort keys by which a list of a resource type can be ordered.
+ */
+public enum SortKey {
+  CREATION_DATE,
+  TRENDING;
```
Comment (jimmordino) [inline service/discussion/src/main/java/com/wikia/discussionservice/enums/SortKey.java]: Are we happy with "TRENDING DESCENDING" or do we want "TRENDING_SCORE"?

Comment (armonr) [inline service/discussion/src/main/java/com/wikia/discussionservice/enums/SortKey.java]: I vote in favor of consistency and predictability for API usage.

**Comment by jimmordino** [comment]
LGTM

**Pull request merged**